codegen/smithy-aws-go-codegen/src/main/java/software/amazon/smithy/aws/go/codegen/EndpointGenerator.java

```diff
@@ -134,6 +136,51 @@ public void run() {
                 });
     }
 
+    private void generateAwsEndpointResolverWrapper(GoWriter writer) {
+        Symbol awsEndpointResolver = SymbolUtils.createValueSymbolBuilder("EndpointResolver", AwsGoDependency.AWS_CORE)
+                .build();
+        Symbol resolverInterface = SymbolUtils.createValueSymbolBuilder(RESOLVER_INTERFACE_NAME).build();
+
+        Symbol wrappedResolverSymbol = SymbolUtils.createPointableSymbolBuilder("wrappedEndpointResolver").build();
+        writer.openBlock("type $T struct {", "}", wrappedResolverSymbol, () -> {
+            writer.write("awsResolver $T", awsEndpointResolver);
+            writer.write("resolver $T", resolverInterface);
+        });
+        writeExternalResolveEndpointImplementation(writer, wrappedResolverSymbol, "w", () -> {
+            writer.openBlock("if w.awsResolver == nil {", "}", () -> writer.write("goto fallback"));
+
+            writer.write("endpoint, err = w.awsResolver.ResolveEndpoint(ServiceID, region)");
+            writer.openBlock("if err == nil {", "}", () -> writer.write("return endpoint, nil"));
+            writer.write("");
+
+            writer.addUseImports(SmithyGoDependency.ERRORS);
+            writer.openBlock("if nf := (&$T{}); !errors.As(err, &nf) {", "}", SymbolUtils.createValueSymbolBuilder("EndpointNotFound",
+                    AwsGoDependency.AWS_CORE).build(), () -> writer.write("return endpoint, err"));
+            writer.write("");
+
+            writer.write("fallback:");
+            writer.openBlock("if w.resolver == nil {", "}", () -> {
+                writer.addUseImports(SmithyGoDependency.FMT);
+                writer.write("return endpoint, fmt.Errorf(\"default endpoint resolver provided was nil\")");
+            });
+            writer.write("return w.resolver.ResolveEndpoint(region, options)");
+        });
+
+        // Generate exported helper for constructing a wrapper around the AWS EndpointResolver type that is compatible
+        // with the clients EndpointResolver interface.
+        writer.writeDocs(String.format("%s returns an EndpointResolver that first delegates endpoint resolution "
+                + "to the awsResolver. If awsResolver returns `aws.EndpointNotFound` error, the resolver will use the "
```

Comment:
Doesn't look like `EndpointNotFound` is defined. The type should probably also have the suffix, `EndpointNotFoundError` for easier discoverability/consistency with others in SDK..

Reply:
It's defined in the `aws` Go package.